packages/snaps-controllers/src/snaps/SnapController.test.ts

```diff
@@ -5766,6 +5849,230 @@ describe('SnapController', () => {
       controller.destroy();
     });
 
+    it('updates a Snap that has dynamic permissions', async () => {
+      const { manifest } = await getMockSnapFilesWithUpdatedChecksum({
+        manifest: getSnapManifest({
+          version: '1.1.0' as SemVerVersion,
+          dynamicPermissions: {
```

Comment:
Can we add a test to ensure that dynamic permissions are revoked if they are removed from the manifest?

Reply:
Added modification for the existing test. It's now checking if there is `PermissionController:revokePermissions` call with permission that was previously granted, but not in initial permissions and also not in dynamic permissions (manifest).